pandas/src/inference.pyx

```diff
@@ -141,12 +143,33 @@ def infer_dtype(object _values):
     values = values.ravel()
 
     # try to use a valid value
-    for i in range(n):
-       val = util.get_value_1d(values, i)
-       if not is_null_datetimelike(val):
-           break
+    for i from 0 <= i < n:
+        val = util.get_value_1d(values, i)
 
-    if util.is_datetime64_object(val) or val is NaT:
+        # same logic as is_null_datetimelike except for integer nat
```

Comment:
@jreback The logic should be updated after a decision in #13467. Could you check?